heartpoints.sh

```diff
@@ -526,6 +605,10 @@ strings_are_equal() { local string1=$1; local string2=$2
     [ "${string1}" = "${string2}" ]
 }
 
+strings_are_not_equal() { local string1=$1; local string2=$2
+    ! strings_are_equal $@
```

Comment:
may be better to make use of the `string1` and `string2` variables rather than `$@`